kpops/cli/pipeline_config.py

```diff
@@ -104,17 +111,22 @@ def customise_sources(
             init_settings: SettingsSourceCallable,
             env_settings: SettingsSourceCallable,
             file_secret_settings: SettingsSourceCallable,
-        ):
+        ) -> tuple[
+            SettingsSourceCallable | Callable[[PipelineConfig], dict[str, Any]], ...
+        ]:
             return (
+                env_settings,
                 init_settings,
                 yaml_config_settings_source,
-                env_settings,
                 file_secret_settings,
             )
 
 
-def yaml_config_settings_source(settings: PipelineConfig) -> dict | list:
+def yaml_config_settings_source(settings: PipelineConfig) -> dict[str, Any]:
     path_to_config = settings.Config.config_path
     if path_to_config.exists():
-        return load_yaml_file(path_to_config)
+        if isinstance(source := load_yaml_file(path_to_config), dict):
+            return source
+        err_msg = f"{path_to_config} must be a mapping."
+        raise TypeError(err_msg)
```

Comment:
nit: would prefer if you skipped the variable assignment if it's not used anywhere else

Reply:
[reason](https://beta.ruff.rs/docs/rules/raw-string-in-exception/)

If you think it isn't justified I can skip the assignment

Reply:
TIL, good point!